File: java/arvore_avl/NoAvl.java
package arvore_avl;

public class NoAvl {
	int item, fator;
	NoAvl esq, dir;

	public NoAvl(int item) {
		this(item, 1, null, null);
	}
	
	public NoAvl(int item, int fator, NoAvl esq, NoAvl dir) {
		this.item = item;
		this.fator = fator;
		this.esq = esq;
		this.dir = dir;
	}
}

File: c/arvore_avl/noAvl.c
#include "noAvl.h"

#include <stdlib.h>

No * newNo(int item) {
	No * no = (No *) malloc(sizeof(No));
	no->item = item;
	no->esq = NULL;
	no->dir = NULL;
	no->fator = 0;
	return no;
}
